core/src/main/java/org/apache/druid/data/input/impl/LocalInputSource.java

```diff
@@ -131,14 +197,15 @@ public boolean equals(Object o)
     if (o == null || getClass() != o.getClass()) {
       return false;
     }
-    LocalInputSource source = (LocalInputSource) o;
-    return Objects.equals(baseDir, source.baseDir) &&
-           Objects.equals(filter, source.filter);
+    LocalInputSource that = (LocalInputSource) o;
+    return Objects.equals(baseDir, that.baseDir) &&
+           Objects.equals(filter, that.filter) &&
+           Objects.equals(files, that.files);
   }
 
   @Override
   public int hashCode()
   {
-    return Objects.hash(baseDir, filter);
+    return Objects.hash(baseDir, filter, files);
   }
```

Comment:
equals and hashCode need unit tests for maintainability.

Reply:
Added.